java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/RelationalSymbolicValue.java

```diff
@@ -101,16 +100,25 @@ public RelationalSymbolicValue(Kind kind) {
 
   @Override
   public List<ProgramState> setConstraint(ProgramState initialProgramState, BooleanConstraint booleanConstraint) {
-    if (!checkRelation(booleanConstraint, initialProgramState)) {
+    if (booleanConstraint == BooleanConstraint.FALSE) {
+      return inverse().setConstraint(initialProgramState, BooleanConstraint.TRUE);
+    }
+    Set<RelationalSymbolicValue> knownRelations = knownRelations(initialProgramState).collect(Collectors.toSet());
```

Comment:
given that method knownRelations is only used here I don't see the point of making it returning a stream. Let it return a set. I am even wondering about making this method part of PS. But that's something else.